app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,46 @@
+class OrdersController < ApplicationController
+  before_action :load_products, :check_cart, only: [:new, :create]
+  def new
+    @order = Order.new
+  end
+
+  def create
+    @order = current_user.orders.build order_params
+    if @order.save
+      add_order_items(@order, @totals)
```

Comment:
nếu add fail thì order có dc rollback ko em., em phải đặt transaction bọc cho toàn bộ thao tác chứ, sao lại để cho mỗi add_order_items dc :v